core/src/main/scala/cats/SemigroupK.scala

```diff
@@ -31,9 +31,9 @@ import simulacrum.{op, typeclass}
   /**
    * Compose two SemigroupK intsances.
    */
-  def composedWith[G[_]: SemigroupK]: SemigroupK[λ[α => F[G[α]]]] =
+  def compose[G[_]: SemigroupK]: SemigroupK[λ[α => F[G[α]]]] =
```

Comment:
That sounds like a good solution to me. It might also mean we want to add something similar to `MonoidK`. I'm okay with that either going into this PR or a follow up one.